sonaranalyzer-dotnet/src/SonarAnalyzer.CSharp/Rules/Hotspots/DoNotHardcodeCredentials.cs

```diff
@@ -40,10 +40,8 @@ public sealed class DoNotHardcodeCredentials : DoNotHardcodeCredentialsBase<Synt
             PropertyAccessTracker = new CSharpPropertyAccessTracker(analyzerConfiguration, rule);
         }
 
-        protected override void Initialize(ParameterLoadingAnalysisContext context)
+        protected override void InitializeSyntax(ParameterLoadingAnalysisContext context)
```

Comment:
I think we can rename `InitializeSyntax` with `RegisterSyntaxNodeActions` or something similar.

Reply:
It's doing what `Initialize` method usually does. Following chain of `ParameterLoadingDiagnosticAnalyzer.Initialize` that prepares parameters and `DoNotHardcodeCredentialsBase.Initialize` that initializes trackers. I think InitializeSyntax is a good name here.

Reply:
It's the word "syntax" which seems confusing to me. The method doesn't act on a syntax, it's registering  some callbacks on a context.